### PR TITLE
fix(templates): prepare InstantSearch template for v4

### PR DESCRIPTION
**Pull request by francoischalifour**
This updates the InstantSearch.js 3 template with a newer syntax that we recommend for InstantSearch.js 4.

This has the benefits of also working with v4, and therefore we won't need to create another template.

### CODE REVIEW

File: src/templates/InstantSearch.js/src/app.js.hbs [hunk 1]
```diff
@@ -34,24 +31,28 @@ search.addWidget(
...
+});
 
 {{#each attributesForFaceting}}
-search.addWidget(
-  instantsearch.widgets.refinementList({
-    container: '#{{this}}-list',
-    attribute: '{{this}}',
-  })
-);
+const {{this}}RefinementList = instantsearch.widgets.refinementList({
```
Comment (Haroenv) [inline src/templates/InstantSearch.js/src/app.js.hbs]: what if the attribute can't be used as a variable? like `my-thing`? We might need to camelcase it, or directly add them in the tree without giving names.

Comment (francoischalifour) [reply src/templates/InstantSearch.js/src/app.js.hbs]: I thought about attributes starting with numbers (not likely) but not about that indeed. I'll camelcase those.

File: src/templates/InstantSearch.js/src/app.js.hbs [hunk 2]
```diff
@@ -34,24 +31,28 @@ search.addWidget(
...
-    container: '#pagination',
-  })
-);
+const pagination = instantsearch.widgets.pagination({
+  container: '#pagination',
+});
+
+search.addWidgets([
+  searchBox,
+  hits,
```
Comment (samouss) [inline src/templates/InstantSearch.js/src/app.js.hbs]: Do we really have to store them? We can easily create anonymous widget.

Comment (francoischalifour) [reply src/templates/InstantSearch.js/src/app.js.hbs]: Sounds good – going to be easier than getting Handlebars to work properly in this environment.

**Pull request merged**